doc/languages-frameworks/python.section.md

```diff
@@ -184,7 +184,15 @@ following are specific to `buildPythonPackage`:
   [`makeWrapper`](#fun-makeWrapper) set `PATH` and `PYTHONPATH` environment variables before calling
   the binary. Additional arguments here can allow a developer to set environment
   variables which will be available when the binary is run. For example,
-  `makeWrapperArgs = ["--set FOO BAR" "--set BAZ QUX"]`.
+  `makeWrapperArgs = ["--set" "FOO" "BAR" "--set" "BAZ" "QUX"]`.
+
+  ::: {.note}
+  When `__structuredAttrs = false`, the attribute `makeWrapperArgs` is passed as a space-separated string to the build script. Developers should use `prependToVar` or `appendToVar` to add arguments to it in build phases, or use `__structuredAttrs = true` to ensure that `makeWrapperArgs` is passed as a Bash array.
+
+  For compatibility purposes,
+  when `makeWrapperArgs` is passed as an IFS-separated string to the build script, the string content is Bash-expanded before concatenated into the `wrapProgram` command. Still, developers should not rely on such behaviours, but use `__structuredAttrs = true` to specify flags containing spaces (e.g. `makeWrapperArgs = [ "--set" "GREETING" "Hello, world!" ]`), or use -pre and -post phases to specify flags with Bash-expansions (e.g. `preFixup = ''makeWrapperArgs+=(--prefix PATH : "$SOME_PATH")`'').
```

Comment:
> The first paragraph talks about a "space separated string", while the second talks about "IFS-separated".

Yes, we should unify them.

> Maybe it would be better to refer to "structuredAttrs = false" in the second paragraph instead?

"`__structuredAttrs = false`" and "`makeWrapperArg` being specified as an IFS-separated string" is semantically different. In current Nixpkgs, `buildPython*`'s `makeWrapperArg` is often specified as an array in the phases and hooks, whereas all those packages are effective `__structuredAttrs = false.` This may require a tree-wide fix in the long run.